dashboard/app/controllers/api/v1/user_school_infos_controller.rb

```diff
@@ -13,21 +13,17 @@ def update_last_confirmation_date
   def update
     return unless school_info_params[:school_id].present? || school_info_params[:country].present?
 
-    new_school_info_params =
-      if school_info_params[:full_address]&.blank?
-        school_info_params.except(:full_address)
-      else
-        school_info_params
-      end
-
-    if new_school_info_params[:country]&.downcase&.eql? 'united states'
-      new_school_info_params[:country] = 'US'
-    end
+    # school_info_params.delete(:full_address) if school_info_params[:full_address]&.blank?
+    e = school_info_params.merge({country: 'US'})
+    e.delete(:full_address) if e[:full_address]&.blank?
+    school_info_params = e
```

Comment:
Using a temporary variable (I had to pull out CSFUN tricks), I was able to re-assign the new values for full address and country.  I am not sure why I cannot do the following:

school_info_params.delete(:full_address) if school_info_params[:full_address]&.blank? (per your recommendation)

and 

if school_info_params[:country]&.downcase&.eql? 'united states'
      school_info_params[:country] = 'US'
end

I tried to check if this has to do with "permit" in school_info_params, but I didn't come up with anything.  Thoughts